pennylane/devices/default_mixed.py

```diff
@@ -449,6 +452,50 @@ def _apply_density_matrix(self, state, device_wires):
             )
             self._state = self._asarray(rho, dtype=self.C_DTYPE)
 
+    def _generate_projector_diagonal(self, wire):
+        n_wires_right = self.num_wires-wire-1
+        n_wires_left = wire
+        id_dim_right = 2**n_wires_right
+        id_dim_left = 2**n_wires_left
+
+        diag = [1]*(id_dim_right) + [0]*(id_dim_right)
+        diag = diag*(id_dim_left)
+
+        return diag
+
+    def _apply_measurement(self, meas_op):
+        zero_projector = self._generate_projector_diagonal(meas_op.wire)
```

Comment:
How about using `qml.Projector` here? It does not seem to allow diagonal usage, but likely would be the best fit in some form. Users could provide their own `qml.Projector` instances for example.

Reply:
I don't think this class works because it only represents projectors on a single computational basis state, not on larger subspaces (we want the subspace generated by fixing a single qubit, not the one generated by fixing all qubits).